numba/typed/dictobject.py

```diff
@@ -1055,24 +1057,252 @@ def impl_iterator_iternext(context, builder, sig, args, result):
 
 
 def build_map(context, builder, dict_type, item_types, items):
-    from numba.typed import Dict
 
-    sig = typing.signature(dict_type)
-    kt, vt = dict_type.key_type, dict_type.value_type
+    if isinstance(dict_type, types.LiteralStrKeyDict):
+        unliteral_tys = [x for x in
+                         dict_type.literal_value.values()]
+        nbty = types.NamedTuple(unliteral_tys,
+                                dict_type.tuple_ty)
+        values = [x[1] for x in items]
+        # replace with make_tuple call?
+        tup = context.get_constant_undef(nbty)
+        literal_tys = [x for x in dict_type.literal_value.values()]
+
+        # this is to deal with repeated keys
+        value_index = dict_type.value_index
+        if value_index is None:
```

Comment:
`value_index` is never `None`

Reply:
It's `None` by default on the type ctor, so if you want to do e.g.
```pycon
In [5]: from numba import types                                                                                                 

In [6]: literal = types.literal                                                                                                 

In [7]: ld = types.LiteralStrKeyDict({literal(k):literal(v) for k, v in {'a':1, 'b':2}.items()})                                

In [8]: ld                                                                                                                      
Out[8]: LiteralStrKey[Dict]({Literal[str](a): Literal[int](1), Literal[str](b): Literal[int](2)})

In [9]: ld.value_index 
```
that's fine, it just puts in the 1:1 map.

Reply:
oh ok